### PR TITLE
Toolstate: Don't block beta week on already broken tools.

### LINKED ISSUE #68458: Broken tool going into toolstate week blocks PRs from landing
Despite Clippy already being broken in toolstate, https://github.com/rust-lang/rust/pull/68432#issuecomment-576955941 failed to land due to Clippy not passing.

We had to fix clippy in https://github.com/rust-lang/rust/pull/68442 so that PRs could start landing again.

It's unclear whether this is intentional or not.

Relevant discussion: https://discordapp.com/channels/442252698964721669/443148319431065610/669558059831394346.

### PR DESCRIPTION
**Pull request by ehuss**
This changes it so that tools are allowed to be broken entering the beta week if they are already broken.  This restores the original behavior before the changes in #66681.

Closes #68458

### CODE REVIEW

**Comment by ehuss** [comment]
No PRs will pass starting Wednesday without this or if all tools aren't fixed by then.

I'll try to get the books fixed before then, but that is cutting it close.  I think Clippy should be good to update, but there is some risk because it will require a coordinated update with Cargo.

**Comment by Mark-Simulacrum** [comment]
This looks correct to me, and does seem like the better behavior: realistically, we have 6 weeks to get tools working on beta branch.

@bors r+

**Comment by Ixrec** [comment]
> This restores the original behavior before the changes in #69332.

Was this meant to link to something else? That PR doesn't touch `toolstate.rs`.

**Comment by ehuss** [comment]
Yea, it was supposed to be #66681.  GitHub's auto-complete for issue numbers is very annoying, and too often "helpfully" picks a random number.

File: src/bootstrap/toolstate.rs [hunk 1]
```diff
@@ -180,11 +203,24 @@ impl Step for ToolStateCheck {
...
+                        .iter()
+                        .find(|ts| ts.tool == *tool)
+                        .expect("latest.json missing tool")
+                        .state();
+                    if state < old_state {
+                        did_error = true;
+                        eprintln!(
+                            "error: Tool `{}` has regressed from {} to {} during beta week.",
+                            tool, old_state, state
+                        );
```
Comment (RalfJung) [inline src/bootstrap/toolstate.rs]: Reading this code, something very odd is going on here... there are two places where the code does "if regressed and in_beta_weak, then fail": it does that here, and it does that again and seemingly independently in `change_toolstate`.

What is going on here? Seems like duplication of the same logic, which usually shouldn't happen.

Comment (RalfJung) [reply src/bootstrap/toolstate.rs]: @ehuss also see https://github.com/rust-lang/rust/pull/69693. Feel free to just include that commit in your PR, then I will close mine.

Comment (ehuss) [reply src/bootstrap/toolstate.rs]: Posted #69705 with the fix.

**Pull request merged**